bootstrap/bootstrap-kubestellar.sh

```diff
@@ -194,67 +248,131 @@ if [ "$folder" == "" ]; then
 fi
 
 # Ensure kcp is installed
-if [ "$(kcp_installed)" == "false" ]; then
-    if [ "$verbose" != "" ]; then
-        echo "Installing kcp..."
+echo "************************************************"
+echo "Ensure kcp is installed"
+echo "************************************************"
+if [ "$(kcp_installed)" == "true" ]; then
+    echo "kcp found in the PATH at '$(which kcp)' ... skip installation."
+else
+    echo "kcp not found in the PATH."
+    if [ "$(kcp_running)" == "true" ]; then
+        echo "kcp process is running already: pid=$(pgrep kcp) ... add kcp folder to the PATH or stop kcp ... exiting."
+        exit 2
     fi
-    bash <(curl -s https://raw.githubusercontent.com/kcp-dev/edge-mc/main/bootstrap/install-kcp-with-plugins.sh) --version $kcp_version --os $os_type --arch $arch_type --ensure-folder  $folder/kcp $verbose $flagx
-    if [[ ! ":$PATH:" == *":$(get_full_path $folder/kcp/bin):"* ]]; then
-        export PATH=$PATH:$(get_full_path $folder/kcp/bin)
+    ensure_folder "$folder/kcp"
+    kcp_folder=$(get_full_path "$folder/kcp")
+    kcp_bin_folder="$kcp_folder/bin"
+    echo "Downloading kcp+plugins $kcp_version $os_type/$arch_type..."
+    kcp_download
+    echo "Installing kcp+plugins into '$kcp_folder'..."
+    kcp_install
+    if [[ ! ":$PATH:" == *":$kcp_bin_folder:"* ]]; then
+        export PATH=$kcp_bin_folder:$PATH
+        echo "Add kcp folder to the PATH: export PATH=\"$kcp_bin_folder:\$PATH\""
+        user_exports="$user_exports"$'\n'"export PATH=\"$kcp_bin_folder:\$PATH\""
     fi
 fi
 
 # Ensure kcp is running
-if [ "$(kcp_running)" == "false" ]; then
+echo "************************************************"
+echo "Ensure kcp is running"
+echo "************************************************"
+if [ "$(kcp_running)" == "true" ]; then
+    echo "kcp process is running already: pid=$(pgrep kcp) ... skip running."
+    if [ "$(kubeconfig_valid)" == "false" ]; then
```

Comment:
The KUBECONFIG envar is not the only relevant factor in kubectl configuration.  There is also the `$HOME/.kube/config` file.  Perhaps a functional test would be better?

Reply:
currently is testing that KUBECONFIG is not empty and points to an existing file.
Additionally, if it is pointing to a non valid file, I expect that kcp_ready and kcp_version would not pass

Reply:
My point is that kubectl does NOT _require_ `$KUBECONFIG` to be set.  If there is a kubeconfig file at `$HOME/.kube/config` then that is enough to make kubectl happy.  If we were being totally good Kubernetes citizens, we would not require more than kubectl requires.